sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/ResourceManagerThrottlingInfo.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.resources.fluentcore.utils;
+
+import com.azure.core.http.HttpHeaders;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The class to collect all throttling info from response header
+ */
+public class ResourceManagerThrottlingInfo {
+    // refer https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/request-limits-and-throttling
+    private static final List<String> COMMON_RATE_LIMIT_HEADERS = Arrays.asList(
+        "x-ms-ratelimit-remaining-subscription-reads",
+        "x-ms-ratelimit-remaining-subscription-writes",
+        "x-ms-ratelimit-remaining-tenant-reads",
+        "x-ms-ratelimit-remaining-tenant-writes",
+        "x-ms-ratelimit-remaining-subscription-resource-requests",
+        "x-ms-ratelimit-remaining-subscription-resource-entities-read",
+        "x-ms-ratelimit-remaining-tenant-resource-requests",
+        "x-ms-ratelimit-remaining-tenant-resource-entities-read"
+    );
+
+    // refer https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/troubleshooting-throttling-errors
+    private static final String RESOURCE_RATE_LIMIT_HEADERS = "x-ms-ratelimit-remaining-resource";
```

Comment:
It would be nice if we can try parse it, regex maybe. We can give user the string, and if we can parse, put value to commonRateLimits.

Reply:
It could be parsed, but it contains two value for different duration `3 min` and `30 min`. What should we use? such as `x-ms-ratelimit-remaining-resource-3min`?

Reply:
Yes, maybe `x-ms-ratelimit-remaining-resource-PutVM3Min` and `x-ms-ratelimit-remaining-resource-PutVM30Min` for `Microsoft.Compute/PutVM3Min;239,Microsoft.Compute/PutVM30Min;1199`.

Or maybe a data structure more complex than key/value pair?